src/utils/MainApi.js

```diff
@@ -0,0 +1,91 @@
+import { BASE_URL } from "./constants";
+
+class Api {
+  constructor({ baseUrl, headers }) {
+    this.baseUrl = baseUrl;
+    this.headers = headers;
+  }
+
+  _processResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`An error just occurred: ${res.status}`);
+    }
+  }
+
+  getCurrentUser(token) {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "GET",
+      headers: {
+        authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    }).then((res) => {
+      return this._processResponse(res);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._processResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.